# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../pearpc/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

crc32.cc: In function 'uint32 ether_crc(size_t, const byte*)':
crc32.cc:109: error: cast from 'const byte*' to 'uint32' loses precision

Go for correctness before we try fix the "fast" code path.

  - Rene Rebe <rene@exactcode.de>

--- pearpc-0.4/src/tools/crc32.cc.vanilla	2009-10-18 13:51:15.000000000 +0200
+++ pearpc-0.4/src/tools/crc32.cc	2009-10-18 13:57:31.000000000 +0200
@@ -97,7 +97,9 @@
 */
 
 // With this macro defined, the function runs about 35% faster, but the code is about 3 times bigger :
+#ifdef __i386__
 #define RUN_FASTER
+#endif
 
 #define DO_CRC(b) crc = (crc >> 8) ^ crc32table[(*(byte*)&crc) ^ (b)]
 
